packages/frontend/src/pages/Send/useL1CanonicalBridge.ts

```diff
@@ -0,0 +1,136 @@
+    return true
+  }
+}
+
+interface Options {
+  customRecipient?: string
+}
+
+export function useL1CanonicalBridge(
+  sourceToken?: Token,
+  sourceTokenAmount?: BigNumber,
+  destNetwork?: Network,
+  estimatedReceived?: BigNumber,
+  txConfirm?: any,
+  options?: Options
+) {
+  const { checkConnectedNetworkId, provider } = useWeb3Context()
+
+  const [l1CanonicalBridge, setL1CanonicalBridge] = useState<CanonicalBridge | undefined>()
+  const [usingNativeBridge, setUsingNativeBridge] = useState(false)
+  const [userSpecifiedBridge, setUserSpecifiedBridge] = useState(false)
+  const { addTransaction } = useTransactionReplacement()
+
+  function selectNativeBridge(val: boolean) {
+    setUsingNativeBridge(val)
+    setUserSpecifiedBridge(true)
+  }
+
+  useEffect(() => {
+    if (userSpecifiedBridge) return
+
+    if (sourceTokenAmount && estimatedReceived && l1CanonicalBridge) {
+      if (!usingNativeBridge && sourceTokenAmount.gt(estimatedReceived)) {
+        setUsingNativeBridge(true)
+      } else if (sourceTokenAmount.lte(estimatedReceived)) {
+        setUsingNativeBridge(false)
+      }
+    }
+
+    return () => setUserSpecifiedBridge(false)
+  }, [sourceTokenAmount?.toString(), estimatedReceived?.toString(), l1CanonicalBridge, destNetwork])
+
+  useEffect(() => {
+    if (!(sourceToken && destNetwork && sourceTokenAmount)) {
+      return setL1CanonicalBridge(undefined)
+    }
+
+    if (sourceToken.chain.chainId !== ChainId.Ethereum) {
+      return setL1CanonicalBridge(undefined)
+    }
+
+    const signer = provider?.getSigner()
+    if (signer) {
+      const canonicalBridge = new CanonicalBridge(
+        NetworkSlug.Mainnet,
+        signer,
+        sourceToken.symbol,
+        destNetwork.slug
+      )
+      setL1CanonicalBridge(canonicalBridge)
+    }
+  }, [provider, sourceTokenAmount?.toString(), sourceToken?.chain.chainId, destNetwork?.slug])
+
+  async function sendL1CanonicalBridge() {
+    if (!(l1CanonicalBridge && sourceTokenAmount)) {
+      return
+    }
+
+    const sourceNetwork = findNetworkById(sourceToken?.chain.chainId.toString()!)
+
+    if (await needsApproval(l1CanonicalBridge, sourceToken, sourceTokenAmount)) {
+      const approveTx = await l1CanonicalBridge.approve(constants.MaxUint256)
+      await approveTx.wait(1)
+    }
+
+    const tx: any = await txConfirm.show({
+      kind: 'depositNativeBridge',
+      inputProps: {
+        customRecipient: options?.customRecipient,
+        source: {
+          amount: toTokenDisplay(sourceTokenAmount, sourceToken?.decimals),
+          token: sourceToken,
+          network: sourceNetwork,
+        },
+        dest: {
+          network: destNetwork,
+        },
+        estimatedReceived: toTokenDisplay(
+          estimatedReceived,
+          sourceToken?.decimals,
+          sourceToken?.symbol
+        ),
+      },
+      onConfirm: async () => {
+        try {
+          const isNetworkConnected = await checkConnectedNetworkId(1)
```

Comment:
we should avoid using hard coded chain ids since it will make it harder to make it work with a testnet. suggestion: use the util to get L1 network id since that reads the network ids from config